### ISSUE TITLE
Ruby sort is not stable -- min_by

### ISSUE THREAD

**Issue by sbudker** [YuFengUofR/tests#32]
https://github.com/YuFengUofR/tests/blob/130114c52db20057f62a1b31187568d5d18785c7/alltest/hao_huang.rb#L462

Although this test checks the output against the real Ruby enumerable, I think this a bad test. Ruby states that its sort methods are not guaranteed to be stable (https://ruby-doc.org/core-2.5.1/Enumerable.html#method-i-sort). I believe Ruby's implementation of min_by uses sort and therefore also gets inconsistent results for cases like this. For example:

<img width="744" alt="screen shot 2018-09-27 at 6 12 23 pm" src="https://user-images.githubusercontent.com/15642848/46177485-f3ff5700-c280-11e8-8c6e-96e34ea3f4ce.png">

In short, I think a test like this is implementation specific since the behavior is not specified for equal elements.

**Comment by hubertlee915** [comment]
@sbudker is right. This is a bad test.
